packages/formatter-html/src/result.ts

```diff
@@ -0,0 +1,239 @@
+    public hasDoc: boolean;
+
+    public constructor(name: string, status: string, url: string, isScanner: boolean) {
+        this.problems = [];
+
+        this.name = name;
+        this.status = status;
+        this.count = 0;
+
+        this.thirdPartyInfo = thirdPartyServices[name];
+
+        if (this.thirdPartyInfo) {
+            this.thirdPartyInfo.link.replace(/%URL%/, url);
+            if (!isScanner) {
+                this.thirdPartyInfo.logo.url = this.thirdPartyInfo.logo.url.substr(1);
+            }
+        }
+
+        this.hasDoc = !hintsWithoutDocs.includes(name);
+    }
+
+    public addProblem(problem: Problem) {
+        this.problems.push(problem);
+        this.count++;
+    }
+}
+
+export class Category {
+    public errors: number;
+    public warnings: number;
+    public hints: Array<Hint>;
+    public name: string;
+    public image: string;
+    public status: string;
+    private cache: Map<string, Hint> = new Map();
+    public url: string;
+    private isScanner: boolean;
+
+    public constructor(name: string, url: string, isScanner: boolean) {
+        this.hints = [];
+        this.name = name;
+
+        this.errors = 0;
+        this.warnings = 0;
+
+        this.image = categoryImages[name.toLowerCase()];
+        this.isScanner = isScanner;
+
+        if (this.image && !isScanner) {
+            this.image = this.image.substr(1);
+        }
+
+        this.status = 'finished';
+        this.url = url;
+    }
+
+    public getHintByName(name: string): Hint | undefined {
+        const lowerCaseName = name.toLowerCase();
+        let hint = this.cache.get(lowerCaseName);
+
+        if (!hint) {
+            hint = this.hints.find((hi: Hint) => {
+                return hi.name.toLowerCase() === lowerCaseName;
+            });
+
+            if (hint) {
+                this.cache.set(lowerCaseName, hint);
+            }
+        }
+
+        return hint;
+    }
+
+    public addHint(name: string): Hint {
+        let hint = this.getHintByName(name);
+
+        if (hint) {
+            return hint;
+        }
+
+        hint = new Hint(name, 'pass', this.url, this.isScanner);
+
+        this.hints.push(hint);
+
+        return hint;
+    }
+
+    public addProblem(problem: Problem) {
+        const hintId = problem.hintId;
+
+        let hint = this.getHintByName(hintId);
+
+        if (!hint) {
+            // All the problems have to have the same severity, so we just need to calculate the status once.
+            const status = problem.severity === Severity.error ? 'error' : 'warning';
+
+            hint = new Hint(hintId, status, this.url, this.isScanner);
+
+            this.hints.push(hint);
+        }
+
+        if (problem.severity === Severity.error) {
+            this.errors++;
+        } else if (problem.severity === Severity.warning) {
+            this.warnings++;
+        }
+
+        hint.addProblem(problem);
+    }
+}
+
+export default class Result {
```

Comment:
Maybe rename to `AnalysisResult` so it's more clear the difference with `HintResult`?

Reply:
yep.